tests/network/provider_migration/libprovider.py

```
@@ -58,3 +80,14 @@ def _check_for_vm_status(vm: vim.VirtualMachine, status: str) -> bool:
         return vm.guest.ipAddress is not None
+
+
+@cache
+def extract_vm_primary_network_data(vm: vim.VirtualMachine) -> tuple[str, str]:
+    for device in vm.config.hardware.device:
+        if isinstance(device, vim.vm.device.VirtualEthernetCard):
+            for net in getattr(vm.guest, "net", []):
+                if net.macAddress == device.macAddress and net.ipAddress:
+                    return net.macAddress, net.ipAddress[0]
+    else:
+        raise IfaceNotFoundError("No network interface found in the VM or no IP address assigned.")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)